cli/src/pcluster/config/cluster_config.py

```diff
@@ -2324,7 +2312,19 @@ def _instance_type_info(self) -> InstanceTypeInfo:
     @property
     def disable_simultaneous_multithreading_manually(self) -> bool:
         """Return true if simultaneous multithreading must be disabled with a cookbook script."""
-        return self.disable_simultaneous_multithreading and self.instance_type_info.default_threads_per_core() > 1
+        return self.disable_simultaneous_multithreading and self._instance_type_info.default_threads_per_core() > 1
+
+    def _instance_id_from_capacity_target(self):
```

Comment:
what about _instance_type_from_capacity_reservation

and modify the docstring accordingly

Reply:
yeah, I'm really not sure why I called it in this way.

Reply:
updated and added unit tests for it.